contracts/contracts/MessageProcessor.sol

```diff
@@ -119,18 +119,9 @@ contract MessageProcessor is Ownable, SnarkCommon, Hasher, CommonUtilities, IMes
       }
     }
 
-    if (
-      !verifyProcessProof(
-        currentMessageBatchIndex,
-        messageRoot,
-        sbCommitment,
-        _newSbCommitment,
-        messageTreeSubDepth,
-        messageTreeDepth,
-        voteOptionTreeDepth,
-        _proof
-      )
-    ) {
+    uint256[] memory publicCircuitInputs = getPublicCircuitInputs(currentMessageBatchIndex, _newSbCommitment);
```

Comment:
you should be able to pass uint8 messageTreeSubDepth, uint8 messageTreeDepth so we don't need to fetch twice, same for these values:  batchEndIndex, _currentMessageBatchIndex

Reply:
same for messageRoot

Reply:
The problem here is related to stack issue I mentioned before + `getPublicCircuitInputs` should be called without additional data passed as params. This is a public method which is also using for proof verification and public inputs check for prover and cli functions.